src/Servers/Kestrel/tools/CodeGenerator/HttpProtocolFeatureCollection.cs

```diff
@@ -12,69 +12,79 @@ public static string GenerateFile()
             var alwaysFeatures = new[]
             {
                 "IHttpRequestFeature",
-                "IHttpRequestBodyDetectionFeature",
                 "IHttpResponseFeature",
                 "IHttpResponseBodyFeature",
-                "IRequestBodyPipeFeature",
-                "IHttpRequestIdentifierFeature",
-                "IServiceProvidersFeature",
-                "IHttpRequestLifetimeFeature",
-                "IHttpConnectionFeature",
                 "IRouteValuesFeature",
-                "IEndpointFeature"
+                "IEndpointFeature",
+                "IServiceProvidersFeature"
             };
 
             var commonFeatures = new[]
             {
-                "IHttpAuthenticationFeature",
-                "IHttpRequestTrailersFeature",
+                "IItemsFeature",
                 "IQueryFeature",
                 "IFormFeature",
+                "IHttpAuthenticationFeature",
+                "IHttpRequestIdentifierFeature",
             };
 
             var sometimesFeatures = new[]
             {
-                "IHttpUpgradeFeature",
-                "IHttp2StreamIdFeature",
-                "IHttpResponseTrailersFeature",
+                "IHttpConnectionFeature",
+                "ISessionFeature",
                 "IResponseCookiesFeature",
-                "IItemsFeature",
+                "IHttpRequestTrailersFeature",
+                "IHttpResponseTrailersFeature",
                 "ITlsConnectionFeature",
-                "IHttpWebSocketFeature",
-                "ISessionFeature",
+                "IHttpUpgradeFeature",
+                "IHttpWebSocketFeature"
+            };
+            var maybeFeatures = new[]
+            {
+                "IHttp2StreamIdFeature",
+                "IRequestBodyPipeFeature",
```

Comment:
Why did this one get downgraded

Reply:
Hmm.. I mistakenly thought Stream and Pipe had been unified under one feature interface; but that looks like its only for Response that that happened

https://github.com/dotnet/aspnetcore/blob/020745cbcb39c4b18733dc3e782ce7cebc70d5be/src/Http/Http.Features/src/IHttpResponseBodyFeature.cs#L19-L24

Reply:
Bumped it back up